app/models/service_container_template.rb

```diff
@@ -0,0 +1,104 @@
+class ServiceContainerTemplate < ServiceGeneric
+  delegate :container_template, :container_manager, :to => :service_template, :allow_nil => true
+
+  # A chance for taking options from automate script to override options from a service dialog
+  def preprocess(action, add_options = {})
+    unless add_options.blank?
+      _log.info("Override with new options:")
+      $log.log_hashes(add_options)
+    end
+
+    save_job_options(action, add_options)
+  end
+
+  def execute(action)
+    opts = get_job_options(action)
+
+    _log.info("Container template provisioning with options:")
+    $log.log_hashes(opts)
+
+    params = process_parameters(opts[:parameters])
+    ManageIQ::Providers::Openshift::ContainerManager::OrchestrationStack.create_stack(container_template, params, opts[:container_project_name])
+    _log.info("Provisioning of container template with ID: [#{id}] name:[#{name}] was initiated.")
+  end
+
+  def check_completed(action)
+    done = stack(action).provider_objects.all? { |obj| obj.resource_category.constantize.find_by(:name => obj.name) }
+    message = done ? "completed" : "not completed yet"
+    [done, message]
+  end
+
+  def refresh(action)
+  end
+
+  def check_refreshed(_action)
+    [true, nil]
+  end
+
+  def on_error(action)
+    _log.info("on_error called for service action: #{action}")
+    container_template.rollback_objects(@created_objects)
+  end
+
+  def stack(action)
+    service_resources.find_by(:name => action, :resource_type => 'OrchestrationStack').try(:resource)
```

Comment:
@lfu @bzwei Are there going to be different stacks based on the `action`?  I would expect there would only ever be one stack connected to `service_resources`.